src/Components/ModelsList/ModelsListItem/ModelsListItem.js

```diff
@@ -25,30 +32,61 @@ class ModelsListItem extends Component {
       .subscribe(({ isHovered }) => this.setState({ isHovered }));
   }
 
+  onNameChangeInput(event) {
+    this.setState({
+      ...this.state,
+      newModelName: event.target.value
+    });
+  }
+
+  applyNameChange() {
+    this.setState({
+      newModelName: DEFAULT_NEW_MODEL,
+      clicked: false
+    });
+    this.props.onApplyChangeName(this.props.model, this.state.newModelName);
+  }
+
+  toggleRenamePanel() {
+    this.setState({
+      ...this.state,
+      clicked: !this.state.clicked
+    });
+  }
+
   render() {
     const { index, model, onRemove } = this.props;
 
     return (
-      <ListItem
-        key={index}
-        button
-        onMouseOver={() => model.setHover(true)}
-        onMouseOut={() => model.setHover(false)}
-        selected={this.state.isHovered}
-      >
-        <ListItemText primary={`${index + 1}) ${model.getName()}`} />
-        <ListItemSecondaryAction>
-          {model.checkbox ? (
-            <Switch
-              defaultChecked={model.checkbox.initialValue}
-              onChange={event => model.checkbox.callback(event.target.checked)}
-            />
-          ) : null}
-          <IconButton aria-label="Delete" onClick={() => onRemove(model)}>
-            <DeleteIcon />
-          </IconButton>
-        </ListItemSecondaryAction>
-      </ListItem>
+      <>
+        <ListItem
+          key={index}
+          button
+          onMouseOver={() => model.setHover(true)}
+          onMouseOut={() => model.setHover(false)}
+          selected={this.state.isHovered}
+        >
+          <ListItemText primary={`${index + 1}) ${model.getName()}`} />
+          <ListItemSecondaryAction>
+            {model.checkbox ? (
+              <Switch
+                defaultChecked={model.checkbox.initialValue}
+                onChange={event => model.checkbox.callback(event.target.checked)}
+              />
+            ) : null}
+            <IconButton aria-label="Edit" onClick={this.toggleRenamePanel.bind(this)}>
+              <EditIcon />
+            </IconButton>
+            <IconButton aria-label="Delete" onClick={() => onRemove(model)}>
+              <DeleteIcon />
+            </IconButton>
+          </ListItemSecondaryAction>
+        </ListItem>
+        {this.state.clicked && <>
+          <TextField fullWidth={true} variant="filled" value={this.state.newModelName} label="Rename the item" onChange={this.onNameChangeInput.bind(this)} />
+          <Button variant="contained" color="secondary" size="small" onClick={this.applyNameChange.bind(this)}>OK</Button>
```

Comment:
Could you bind `applyNameChange` in constructor? (to avoid binding during each re-render)

Reply:
Fixed